### PR TITLE
Update setup.sh to allow template files, add githubissues

### LINKED ISSUE #125: [bug] Project and Region needs to be specified
After following the instructions [on this page](https://cloud.google.com/build/docs/configuring-notifications/automate), running the `setup.sh` script failed for the different reasons I've specified below along with how I resolved them.  Hopefully they'll help the maintainers improve the script and help others who might hit the same stumbling blocks.

### Missing Documentation
The script does not set the current project for you.  Before running the `setup.sh` script, set your current project using `gcloud config set project <project-id>`.

**Recommended Solution**
Add a step to the instructions to set the project before running the `setup.sh` script.  Alternatively, set the project automatically within the `setup.sh` script.

### Region Must Be Specified
The `deploy_notifier()` function should specify the region, otherwise it can return an error if no default region is configured.  For my needs I just hard coded the region `us-central1` into the deployment command, but that should be a variable passed as an argument.

```
deploy_notifier() {
  gcloud run deploy "${SERVICE_NAME}" \
    --image="${IMAGE_PATH}" \
    --no-allow-unauthenticated \
    --region us-central1 \
    --update-env-vars="CONFIG_PATH=${DESTINATION_CONFIG_PATH},PROJECT_ID=${PROJECT_ID}" ||
    fail "failed to deploy notifier service -- check service logs for configuration error"
}
```

**Recommended Solution**
Pass the region as another argument when running `setup.sh`.

### Pub/Sub Invalid Push Url
The build failed a third time with the error:
`ERROR: Failed to create subscription [projects/<project-id>/subscriptions/slack-subscription]: You have passed an invalid argument to the service (argument=PushConfig::push_endpoint).`

The SERVICE_URL value was empty because the `gcloud run services describe` command was called without a region specified, so it was returning an error and pushing an empty value to the --push-url argument.  My quick solution was to simply hard code the SERVICE_URL value on line 111 with the deployed Cloud Run service's URL and it worked for my needs.

**Recommended Solution**
Ensure the region parameter is passed to the `gcloud run services describe` command, like this:
```
  SERVICE_URL=$(gcloud run services describe "${SERVICE_NAME}" \
    --region="${REGION}" 
    --format="value(status.url)")
```

### PR DESCRIPTION
**Pull request by glasnt**
### Features

Add template file flag, change optional params to flags

 - Secret name is now `-s`, Template file is now `-t`, both are optional
(though the `uri` in the template file needs to be set `${PROJECT_ID}-notifiers-config` explicitly as that where the config will end up.)
 - Mandatory fields are still the notifier type and the local path to the config file. 

Add github issues to setup

  - With the template file changes, githubissues can now be added as an automated notifier type. 

### Fixes

Resolve #125 

 - Getting project ID from gcloud returns empty (as of gcloud 393.0.0), so update check to validate empty or "(unset)" as no project ID. 
 - Get region from Cloud Run settings, similarly erroring if not set. 
 
### Misc

* Show example of conditional template in githubissues, and move Trigger ID to this area (not all builds have trigger IDs)

### CODE REVIEW

**Comment by glasnt** [comment]
friendly ping @Aric1088

**Comment by Aric1088** [comment]
LGTM! Next time request a review from Github or send a ping so I get the notification to do the code review!

File: githubissues/githubissues.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {
-    "title": "Cloud Build [{{.Build.ProjectId}}] {{.Build.BuildTriggerId}}: {{.Build.Status}}",
-    "body": "Cloud Build {{.Build.ProjectId}} {{.Build.BuildTriggerId}} status: **{{.Build.Status}}**\n\n[View Logs]({{.Build.LogUrl}})"
-}
-
+    "title": "Cloud Build [{{.Build.ProjectId}}]: {{.Build.Status}}",
+    "body": "Cloud Build {{.Build.ProjectId}} {{.Build.Id}} status: **{{.Build.Status}}**\n\n{{if .Build.BuildTriggerId}}Trigger ID: {{.Build.BuildTriggerId}}{{end}}\n\n[View Logs]({{.Build.LogUrl}})"
+}
```
Comment (Aric1088) [inline githubissues/githubissues.json:4]: add new line

**Review by Aric1088 — CHANGES_REQUESTED** [review]

**Comment by glasnt** [comment]
Change made. I don't have permissions to request, hence the ping :)

**Comment by Aric1088** [comment]
oh I see, that is interesting. I will look into resolving that. In the meantime, feel free to ping on chat or email!

**Review by Aric1088 — APPROVED** [review]
LGTM

**Pull request merged**